### PR TITLE
BaSP-Tg 27 test employee put delete

### PR DESCRIPTION
**Pull request by Ericks19**
Worked testing the endpoints of employee whit methods put and delete

### CODE REVIEW

File: src/test/employees.test.js [hunk 1]
```diff
@@ -1 +1,104 @@
...
+    const response = await request(app).put(`/employees/${employeeIdPeter}`).send(mockedEmployeeModified);
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data.firstName).toBe('Carlos');
+    expect(response.body.data.lastName).toBe('Guevara');
+    expect(response.body.data.email).toBe('CarlosGuevara@gmail.com');
+    expect(response.body.data.password).toBe('passwordreseguro');
+    expect(response.body.data.dni).toBe(12345678);
+    expect(response.body.data.phone).toBe(1168542425);
+    expect(response.body.data.location).toBe('Miame');
```
Comment (MaicolFreire001) [inline src/test/employees.test.js:82]: You could make a mocked employee with dni and phone being numbers so you could check the entire body at once

File: src/test/employees.test.js [hunk 2]
```diff
@@ -1 +1,104 @@
...
+    expect(response.body.message).toBe(`Modified employee with id ${employeeIdPeter}`);
+  });
+  test('should return status code 400', async () => {
+    const invalidId = 254;
```
Comment (MaicolFreire001) [inline src/test/employees.test.js:86]: invalidId could be a global variable because is used more than once

**Review by MaicolFreire001 — CHANGES_REQUESTED** [review]

File: src/test/employees.test.js [hunk 3]
```diff
@@ -1 +1,100 @@
...
+  firstName: 'Carlos',
+  lastName: 'Hills',
+  email: '',
+  password: 'Axhvbhd7844',
+  dni: '30457895',
+  phone: '1168542485',
+  location: 'Montana',
+};
+
+const employeeIdPeter = '63531244ec6456efd12685ef';
```
Comment (nicolasblasco) [inline src/test/employees.test.js:47]: We could rename the variable to firstEmployeeIdFromSeed and get the id like this employeesSeed[0]._id

File: src/test/employees.test.js [hunk 4]
```diff
@@ -1 +1,100 @@
...
+const employeeIdErnesto = '63573014db9ebdd9d24e7b93';
```
Comment (nicolasblasco) [inline src/test/employees.test.js:48]: We could rename the variable to secondEmployeeIdFromSeed and get the id like this employeesSeed[1]._id

File: src/test/employees.test.js [hunk 5]
```diff
@@ -1 +1,100 @@
...
+    const idThatNotexist = '62731244ec6456efd12685ef';
+    const response = await request(app).put(`/employees/${idThatNotexist}`).send(mockedEmployee);
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.message).toBe(`Couldn't find employee with id ${idThatNotexist}`);
+  });
+  test('should return status code 400 validate error', async () => {
+    const response = await request(app).put(`/employees/${employeeIdPeter}`).send(mockedEmployeeInvalid);
+    expect(response.status).toBe(400);
+    expect(response.body.message[0].message).toBe('email required');
```
Comment (nicolasblasco) [inline src/test/employees.test.js:98]: We should also test the error of the body

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

File: src/test/employees.test.js (hunk 5 above)
Comment (Ericks19) [reply src/test/employees.test.js:98]: The error 400 from the validations doesnt throw any error I guess it gives a message and data only

**Review by nicolasblasco — DISMISSED** [review]

**Review by MaicolFreire001 — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**